### PR TITLE
create tests for mp-b2b-abstract-product-prices

### PR DESCRIPTION
**Pull request by m-alama**

### CODE REVIEW

File: tests/api/mp_b2b/glue/abstract_product_endpoints/abstract_product_prices/positive.robot [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+    And Each array element of array in response should contain nested property:    [included]    [attributes][prices][0][volumePrices]    quantity
+    And Response body parameter should be:    [included][0][attributes][prices][0][priceTypeName]    DEFAULT
+    And Response body parameter should be:    [included][0][attributes][prices][0][grossAmount]    789
+    And Response body parameter should be:    [included][0][attributes][prices][0][currency][code]    ${currency_code_eur}
+    And Response body parameter should be:    [included][0][attributes][prices][0][currency][name]    ${currency_name_eur}
+    And Response body parameter should be:    [included][0][attributes][prices][0][currency][symbol]    ${currency_symbol_eur}
+
+
+
+Abstract_prices_original_price
```
Comment (vadym-uskov) [inline tests/api/mp_b2b/glue/abstract_product_endpoints/abstract_product_prices/positive.robot:125]: Please rename to 'Get_abstract_product_with_original_price'

File: tests/api/mp_b2b/glue/abstract_product_endpoints/abstract_product_prices/positive.robot [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+ENABLER
+    TestSetup
+    
+Get_abstract_prices_detault_only
+    [Setup]    Run Keywords    I get access token for the customer:    ${yves_user_email}
+    ...    AND    I set Headers:    Content-Type=${default_header_content_type}    Authorization=${token}  
+    When I send a GET request:    /abstract-products/${abstract_available_product_with_3_concretes}/abstract-product-prices
+    Then Response status code should be:    200
+    And Response reason should be:    OK
+    And Response header parameter should be:    Content-Type    ${default_header_content_type}
```
[the remark this answers is not in the document]
Comment (m-alama) [reply tests/api/mp_b2b/glue/abstract_product_endpoints/abstract_product_prices/positive.robot:17]: Done

File: tests/api/mp_b2b/glue/abstract_product_endpoints/abstract_product_prices/positive.robot (hunk 1 above)
Comment (m-alama) [reply tests/api/mp_b2b/glue/abstract_product_endpoints/abstract_product_prices/positive.robot:125]: Done

File: tests/api/mp_b2b/glue/abstract_product_endpoints/abstract_product_prices/positive.robot [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+    [Setup]    Run Keywords    I get access token for the customer:    ${yves_user_email}
+    ...    AND    I set Headers:    Content-Type=${default_header_content_type}    Authorization=${token}  
+    When I send a GET request:    /abstract-products/${abstract_available_product_with_3_concretes}?include=abstract-product-prices
+    Then Response status code should be:    200
+    And Response reason should be:    OK
+    And Response header parameter should be:    Content-Type    ${default_header_content_type}
+    And Response should contain the array larger than a certain size:    [included]    0
+    And Response should contain the array larger than a certain size:    [data][relationships]    0
+    And Response body parameter should not be EMPTY:    [data][relationships]
+    And Response body parameter should not be EMPTY:    [data][relationships][abstract-product-prices]
```
[the remark this answers is not in the document]
Comment (m-alama) [reply tests/api/mp_b2b/glue/abstract_product_endpoints/abstract_product_prices/positive.robot:41]: Done

File: tests/api/mp_b2b/glue/abstract_product_endpoints/abstract_product_prices/positive.robot [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+
+Get_abstract_product_with_include_abstract_product_prices_only_default
+    [Setup]    Run Keywords    I get access token for the customer:    ${yves_user_email}
+    ...    AND    I set Headers:    Content-Type=${default_header_content_type}    Authorization=${token}  
+    When I send a GET request:    /abstract-products/${abstract_available_product_with_3_concretes}?include=abstract-product-prices
+    Then Response status code should be:    200
+    And Response reason should be:    OK
+    And Response header parameter should be:    Content-Type    ${default_header_content_type}
+    And Response should contain the array larger than a certain size:    [included]    0
+    And Response should contain the array larger than a certain size:    [data][relationships]    0
```
[the remark this answers is not in the document]
Comment (m-alama) [reply tests/api/mp_b2b/glue/abstract_product_endpoints/abstract_product_prices/positive.robot:39]: Done

File: tests/api/mp_b2b/glue/abstract_product_endpoints/abstract_product_prices/positive.robot [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+    And Each array element of array in response should contain nested property:    [included]    [attributes][prices]    priceTypeName
+    And Response body parameter should be:    [included][0][attributes][prices][0][priceTypeName]    DEFAULT
+    And Response body parameter should be:    [included][0][attributes][prices][0][grossAmount]    26797
+    And Response body parameter should be:    [included][0][attributes][prices][0][currency][code]    ${currency_code_eur}
+    And Response body parameter should be:    [included][0][attributes][prices][0][currency][name]    ${currency_name_eur}
+    And Response body parameter should be:    [included][0][attributes][prices][0][currency][symbol]    ${currency_symbol_eur}
+    
+
+
+Abstract_volume_prices
```
[the remark this answers is not in the document]
Comment (m-alama) [reply tests/api/mp_b2b/glue/abstract_product_endpoints/abstract_product_prices/positive.robot:63]: Done

File: tests/api/mp_b2b/glue/abstract_product_endpoints/abstract_product_prices/positive.robot [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+    And Response header parameter should be:    Content-Type    ${default_header_content_type}
+    And Response should contain the array larger than a certain size:    [included]    0
+    And Response should contain the array larger than a certain size:    [data][relationships]    0
+    And Response body parameter should not be EMPTY:    [data][relationships]
+    And Response body parameter should not be EMPTY:    [data][relationships][abstract-product-prices]
+    And Each array element of array in response should contain property:    [data][relationships][abstract-product-prices][data]    type
+    And Each array element of array in response should contain property:    [data][relationships][abstract-product-prices][data]    id
+    And Response body parameter should not be EMPTY:    [included]
+    And Each array element of array in response should contain property:    [included]    type
+    And Each array element of array in response should contain property:    [included]    id
```
[the remark this answers is not in the document]
Comment (m-alama) [reply tests/api/mp_b2b/glue/abstract_product_endpoints/abstract_product_prices/positive.robot:105]: Done

File: tests/api/mp_b2b/glue/abstract_product_endpoints/abstract_product_prices/positive.robot [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+    When I send a GET request:    /abstract-products/${abstract_available_product_with_3_concretes}?include=abstract-product-prices
+    Then Response status code should be:    200
+    And Response reason should be:    OK
+    And Response header parameter should be:    Content-Type    ${default_header_content_type}
+    And Response should contain the array larger than a certain size:    [included]    0
+    And Response should contain the array larger than a certain size:    [data][relationships]    0
+    And Response body parameter should not be EMPTY:    [data][relationships]
+    And Response body parameter should not be EMPTY:    [data][relationships][abstract-product-prices]
+    And Each array element of array in response should contain property:    [data][relationships][abstract-product-prices][data]    type
+    And Each array element of array in response should contain property:    [data][relationships][abstract-product-prices][data]    id
```
[the remark this answers is not in the document]
Comment (m-alama) [reply tests/api/mp_b2b/glue/abstract_product_endpoints/abstract_product_prices/positive.robot:43]: Done

File: tests/api/mp_b2b/glue/abstract_product_endpoints/abstract_product_prices/positive.robot [hunk 8]
```diff
@@ -0,0 +1,145 @@
+*** Settings ***
+Suite Setup       SuiteSetup
+Test Setup        TestSetup
+Resource    ../../../../../../resources/common/common_api.robot
+Default Tags    glue
+
+*** Test Cases ***
```
[the remark this answers is not in the document]
Comment (m-alama) [reply tests/api/mp_b2b/glue/abstract_product_endpoints/abstract_product_prices/positive.robot:7]: Done

File: tests/api/mp_b2b/glue/abstract_product_endpoints/abstract_product_prices/positive.robot [hunk 9]
```diff
@@ -0,0 +1,137 @@
...
+    And Each array element of array in response should contain nested property:    [included]    attributes    price
+    And Each array element of array in response should contain nested property:    [included]    attributes    prices
+    And Each array element of array in response should contain nested property:    [included]    [attributes][prices]    priceTypeName
+    And Each array element of array in response should contain nested property:    [included]    [attributes][prices][0][volumePrices]    grossAmount
+    And Each array element of array in response should contain nested property:    [included]    [attributes][prices][0][volumePrices]    netAmount
+    And Each array element of array in response should contain nested property:    [included]    [attributes][prices][0][volumePrices]    quantity
+    And Response body parameter should be:    [included][0][id]    ${abstract_product_with_volume_prices}
+    And Response body parameter should be:    [included][0][attributes][prices][0][priceTypeName]    DEFAULT
+    And Response body parameter should be:    [included][0][attributes][prices][0][grossAmount]    789
```
Comment (vadym-uskov) [inline tests/api/mp_b2b/glue/abstract_product_endpoints/abstract_product_prices/positive.robot:111]: Please use 'Response body parameter should be greater than:' keyword to check grossAmount

**Review by OKrivtsova — APPROVED** [review]

**Pull request merged**